src/modules/commander/companion_process_status.h

```diff
@@ -0,0 +1,98 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013, 2014 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file companion_process_status.h
+ * Routine to determine health of companion processes
+ */
+
+#ifndef COMPANION_PROCESS_STATUS_H_
+#define COMPANION_PROCESS_STATUS_H_
+
+#include <stdint.h>
+#include <uORB/uORB.h>
+#include <uORB/topics/companion_process_status.h>
+#include <px4_defines.h>
+#include <px4_posix.h>
+#include <px4_time.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <cmath>
+#include <uORB/Subscription.hpp>
+#include <drivers/drv_hrt.h>
+#include <systemlib/mavlink_log.h>
+#include <cstring>
+
+enum ProcessType {UNDEFINED, AVOIDANCE, VIO};  //used as indexes for _companion_process_status arrays
+enum MAV_COMPONENT {MAV_COMP_ID_OBSTACLE_AVOIDANCE = 196, MAV_COMP_ID_VISUAL_INERTIAL_ODOMETRY = 197};  //see mavlink::common::MAV_COMPONENT
```

Comment:
You don't need this enum as you can just use the Mavlink `MAV_COMPONENT` one.

Reply:
@bkueng recommended making a separate one so that we do not need to include the mavlink header ouside of the mavlink module. I would be fine with both solutions, which one do you like better?

Reply:
Beats recommendation is the standard pattern with mavlink in px4. Proper encapsulation of dependencies.

Reply:
@thomasgubler sounds reasonable.